### PR TITLE
Use platform-specific back icon in Scaffold

### PR DESCRIPTION
**Pull request by cbracken**

### CODE REVIEW

**Comment by cbracken** [comment]
Addresses #4955

File: packages/flutter/lib/src/material/scaffold.dart [hunk 1]
```diff
@@ -614,8 +614,11 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
       } else {
         _shouldShowBackArrow ??= Navigator.canPop(context);
         if (_shouldShowBackArrow) {
+          IconData backIcon = Theme.of(context).platform == TargetPlatform.iOS
+              ? Icons.arrow_back_ios
+              : Icons.arrow_back;
```
Comment (abarth) [inline packages/flutter/lib/src/material/scaffold.dart]: Please use a `switch` statement.  That will make the analyze complain when we add a new target platform and someone will need to think about which icon it should have.

**Review by abarth — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/material/scaffold.dart (hunk 1 above)
Comment (cbracken) [inline packages/flutter/lib/src/material/scaffold.dart]: Good call -- done. Added a test for Fuchsia too.

**Comment by abarth** [comment]
LGTM

**Pull request merged**